src/heroku-keepalive.coffee

```diff
@@ -38,16 +44,36 @@ module.exports = (robot) ->
   if keepaliveInterval > 0.0
     robot.herokuKeepaliveIntervalId = setInterval =>
       robot.logger.info 'keepalive ping'
-      robot.http("#{keepaliveUrl}heroku/keepalive").post() (err, res, body) =>
-        if err?
-          robot.logger.info "keepalive pong: #{err}"
-          robot.emit 'error', err
+
+      if process.env.HUBOT_HEROKU_WAKEUP_TIME && process.env.HUBOT_HEROKU_SLEEP_TIME
+        wakeUpTime = process.env.HUBOT_HEROKU_WAKEUP_TIME.split(':')
+        sleepTime  = process.env.HUBOT_HEROKU_SLEEP_TIME.split(':')
+
+        wakeUpOffset = 60 * wakeUpTime[0]  + 1 * wakeUpTime[1]
+        sleepOffset  = 60 * sleepTime[0]   + 1 * sleepTime[1]
+
+        now = new Date()
+        nowOffset    = 60 * now.getHours() + now.getMinutes()
+
+        if (nowOffset >= wakeUpOffset && nowOffset < sleepOffset)
+          callKeepAlive()
         else
-          robot.logger.info "keepalive pong: #{res.statusCode} #{body}"
+          robot.logger.info "Skipping keep alive, time to rest"
+      else
+        callKeepAlive()
```

Comment:
Yes, it probably should be. Since all paid plans never sleep, this should only be used for their free tier. If being used on the free tier, this would basically just let Heroku determine when it puts their app to sleep. Will update shortly.